incubator/hnc/pkg/stats/stats.go

```diff
@@ -72,22 +84,32 @@ func StopObjReconcile(gvk schema.GroupVersionKind) {
 // WriteNamespace updates stats when writing namespace instance.
 func WriteNamespace() {
 	stats.namespaceWrites.incr()
+
+	m := int64(stats.namespaceWrites)
+	recordMetrics(m, namespaceWritesTotal)
 }
 
 // WriteHierConfig updates stats when writing hierarchyConfig instance.
 func WriteHierConfig() {
 	stats.hierConfigWrites.incr()
-	recordMetricsInt64(stats.hierConfigWrites, hierConfigWritesTotal)
+
+	m := int64(stats.hierConfigWrites)
+	recordMetrics(m, hierConfigWritesTotal)
```

Comment:
Are you always starting with an `int`? If so, why not have `recordMetrics` accept `int` and cast it inside the function?

Reply:
The temp max values are using int64, the same as the metrics type. Since `recordMetrics()` also records the int64 max value, I moved the cast outside of the function.

I do feel it's verbose too. I will change the max types from int64 to counter, so they can all be cast inside the function.